docs/src/proposals/comprehensive-compute-fees-review.md

```diff
@@ -0,0 +1,164 @@
+---
+title: Comprehensive Compute Fees
+---
+
+## Motivation
+
+The current fee structure lacks a comprehensive account of the work required by
+a validator to process a transaction.  The fee structure is only based on the
+number of signatures in a transaction but is meant to account for the work that
+the validator must perform to validate each transaction.  The validator performs
+a lot more user-defined work than just signature verification.  Processing a
+transaction typically includes signature verifications, account locking, account
+loading, and instruction processing.
+
+## Proposed Solution
+
+The following solution does not specify what native token costs are to be
+associated with the new fee structure.  Instead, it sets the criteria and
+provides the knobs that a cost model can use to determine those costs.
+
+### Fee
+
+The goal of the fees are to prevent the abuse of on-chain compute resources when
+processing  a transaction.  Each of the fee categories below will be represented
+as a compute unit cost that, when added together, encompasses the entire cost of
+processing the transaction.  By calculating the total cost of the transaction,
+the runtime can charge a more representative fee and make better transaction
+scheduling decisions.
+
+A fee will be calculated based on:
+
+1. Number of signatures
+   - Fixed rate per signature
+2. Number of write locks
+   - Fixed rate per writable account
+3. Data byte cost
```

Comment:
does fee structure need to be so complex? what if side effects are measured instead, like number of CPU clock cycles a thread has spent executing the transaction (normalized by frequency)? then it is easy to understand and estimate on the user side

Reply:
Fee calculations must be deterministic, what did you have in mind for measurements?